frontend/explorer/public/js/hash.js

```diff
@@ -1209,25 +1209,100 @@ function populateHashPage(hash, explorerHash) {
 		appendHeading(infoBody, 'Hash: ' + hash);
 		appendExplorerBlock(infoBody, explorerHash.block);
 	} else if (hashType === "transactionid") {
+		appendNavigationMenuTransaction(explorerHash.transaction);
 		appendHeading(infoBody, 'Hash Type: Transaction ID');
 		appendHeading(infoBody, 'Hash: ' + hash);
 		appendTransactionStatistics(infoBody, explorerHash.transaction, explorerHash.unconfirmed!==true);
 		appendRawTransaction(infoBody, explorerHash.transaction.rawtransaction);
 	} else if (hashType === "unlockhash") {
+		appendNavigationMenuUnlockHash(hash);
 		appendHeading(infoBody, 'Hash Type: Unlock Hash');
 		appendHeading(infoBody, 'Hash: ' + hash);
 		appendUnlockHashTables(infoBody, hash, explorerHash);
 	} else if (hashType === "coinoutputid") {
+		appendNavigationMenuCoinOutput(explorerHash, hash);
 		appendHeading(infoBody, 'Hash Type: Coin Output ID');
 		appendHeading(infoBody, 'Hash: ' + hash);
 		appendCoinOutputTables(infoBody, hash, explorerHash);
 	} else if (hashType === "blockstakeoutputid") {
+		appendNavigationMenuBlockstakeOutput(explorerHash.transactions, hash);
 		appendHeading(infoBody, 'Hash Type: BlockStake Output ID');
 		appendHeading(infoBody, 'Hash: ' + hash);
 		appendBlockStakeOutputTables(infoBody, hash, explorerHash);
 	}
 }
 
+// appendNavigationMenuTranscaction adds the transaction link to the top navigation menu
+function appendNavigationMenuTransaction(explorerTransaction) {
+	appendNavigationMenuBlock(explorerTransaction);
+	var navigation = document.getElementById('nav-links');
+	var transactionSpan = document.createElement('span');
+	navigation.appendChild(document.createTextNode(' > '));
+	navigation.appendChild(transactionSpan);
+	linkHash(transactionSpan, explorerTransaction.id, 'Transaction');
+}
+
+// appendNavigationMenuCoinOuput adds the coin output link to the top navigation menu
+function appendNavigationMenuCoinOutput(explorerHash, hash) {
+	//Coin Ouput
+	if (explorerHash.transactions != null) {
+		for (var i = 0; i < explorerHash.transactions.length; i++) {
+			for (var j = 0; j < explorerHash.transactions[i].coinoutputids.length; j++) {
+				if (explorerHash.transactions[i].coinoutputids[j] == hash) {
+					appendNavigationMenuTransaction(explorerHash.transactions[i]);
+					var navigation = document.getElementById('nav-links');
+					var outputSpan = document.createElement('span');
+					navigation.appendChild(document.createTextNode(' > '));
+					navigation.appendChild(outputSpan);
+					linkHash(outputSpan, explorerHash.transactions[i].coinoutputids[j], 'Coin Output');
+				}
```

Comment:
add a `return` statement here, this will give you 2 benefits:

1. it will stop looping as soon as it found it
2. you can remove the else if (see comment below)

Reply:
Added return statement
Removed else if